src/chapter01/question1.2/HashMapStringPermutationChecker.php

```diff
@@ -2,31 +2,16 @@
 
 class HashMapStringPermutationChecker {
     public static function isPermutation($str1, $str2) {
-        $length = strlen($str1);
-        if ($length !== strlen($str2)) {
+        if (strlen($str1) !== strlen($str2)) {
             return false;
         }
-        $charCounts1 = [];
-        $charCounts2 = [];
-        for ($i=0; $i<$length; $i++) {
-            $char1 = $str1[$i];
-            if (!array_key_exists($char1, $charCounts1)) {
-                $charCounts1[$char1]=1;
-            } else {
-                $charCounts1[$char1]++;
-            }
-            $char2 = $str2[$i];
-            if (!array_key_exists($char2, $charCounts2)) {
-                $charCounts2[$char2]=1;
-            } else {
-                $charCounts2[$char2]++;
-            }
-        }
+        $charCounts1 = array_count_values(str_split($str1));
+        $charCounts2 = array_count_values(str_split($str2));
```

Comment:
Although [`array_count_values()`](http://php.net/manual/en/function.array-count-values.php) is arguably more idiomatic php, since this is primarily meant to illustrate an algorithm, I prefer the explicit inclusion of the algorithm's steps, even if it more verbose.